File: communicator/message_type.py
from enum import Enum

class MessageType(Enum):
    CONTEXT = "context"
    FUNCTIONALITY = "functionality"
    ACTION = "action"
    TASK = "task"
    USER = "user"
    TASK_INITIATION = "task_initiation"
    TASK_TERMINATION = "task_termination"
    
    def __str__(self):
        return self.value
    
    @classmethod
    def from_string(cls, value):
        for m in cls:
            if m.value == value:
                return m
        raise ValueError(f"No such enum constant with specified value: {value}")

File: lapras-middleware/src/main/java/kr/ac/kaist/cdsn/lapras/communicator/MessageType.java
package kr.ac.kaist.cdsn.lapras.communicator;

/**
 * Created by Daekeun Lee on 2016-11-14.
 */
public enum MessageType {
    CONTEXT("context"),
    FUNCTIONALITY("functionality"),
    ACTION("action"),
    TASK("task"),
    USER("user"),
    TASK_INITIATION("task_initiation"),
    TASK_TERMINATION("task_termination");

    String name;
    MessageType(String name) {
        this.name = name;
    }

    public String toString(){
        return name;
    }

    public static MessageType fromString(String value) {
        for(MessageType m : MessageType.values()) {
            if(m.toString().equals(value))
                return m;
        }
        throw new IllegalArgumentException("No such enum constant with specified value");
    }
}
